clippy_lints/src/unused_async.rs

```diff
@@ -42,21 +42,43 @@ declare_lint_pass!(UnusedAsync => [UNUSED_ASYNC]);
 struct AsyncFnVisitor<'a, 'tcx> {
     cx: &'a LateContext<'tcx>,
     found_await: bool,
+    /// Also keep track of `await`s in nested async blocks so we can mention
+    /// it in a note
+    found_await_in_async_block: bool,
```

Comment:
I'm not against doing this, but it should be keeping track of the span as well. Just the first occurrence is fine.

Reply:
Alright, done. Though not sure exactly if this is how you meant it.